contentcuration/contentcuration/frontend/channelList/views/Channel/StudioMyChannels.vue

```diff
@@ -0,0 +1,533 @@
+                  icon="optionsVertical"
+                  appearance="flat-button"
+                  :data-testid="`dropdown-button-${index}`"
+                  @click="openDropDown(channel, index)"
+                  @click.stop
+                >
+                  <template #menu>
+                    <KDropdownMenu
+                      :hasIcons="true"
+                      :options="dropDownArr"
+                      @select="option => selectedItem(option, channel)"
+                    />
+                  </template>
+                </KIconButton>
+              </div>
+            </div>
+          </template>
+        </KCard>
+      </KCardGrid>
+    </div>
+    <KModal
+      v-if="deleteDialog"
+      :title="canEdit ? $tr('deleteTitle') : $tr('removeTitle')"
+      :submitText="canEdit ? $tr('deleteChannel') : $tr('removeBtn')"
+      :cancelText="$tr('cancel')"
+      data-testid="delete-modal"
+      @submit="handleDelete"
+      @cancel="deleteDialog = false"
+    >
+      {{ canEdit ? $tr('deletePrompt') : $tr('removePrompt') }}
+    </KModal>
+    <ChannelTokenModal
+      v-if="selectedChannel && selectedChannel.published"
+      v-model="tokenDialog"
+      data-testid="copy-modal"
+      :channel="selectedChannel"
+      @copied="trackTokenCopy"
+    />
+  </div>
+
+</template>
+
+
+<script>
+
+  import { mapGetters, mapActions } from 'vuex';
+  import orderBy from 'lodash/orderBy';
+  import useKResponsiveWindow from 'kolibri-design-system/lib/composables/useKResponsiveWindow';
+  import { RouteNames } from '../../constants';
+  import ChannelStar from './ChannelStar';
+  import { ChannelListTypes } from 'shared/constants';
+  import ChannelTokenModal from 'shared/views/channel/ChannelTokenModal';
+  import Languages from 'shared/leUtils/Languages';
+
+  export default {
+    name: 'StudioMyChannels',
+    components: {
+      ChannelStar,
+      ChannelTokenModal,
+    },
+    setup() {
+      const { windowIsSmall, windowIsMedium, windowIsLarge, windowBreakpoint } =
+        useKResponsiveWindow();
+      return {
+        windowIsSmall,
+        windowIsMedium,
+        windowIsLarge,
+        windowBreakpoint,
+      };
+    },
+    data() {
+      return {
+        loading: false,
+        tokenDialog: false,
+        deleteDialog: false,
+        selectedChannel: {
+          published: false,
+        },
+        dropDownArr: [],
+      };
+    },
+    computed: {
+      ...mapGetters('channel', ['channels']),
+      listChannels() {
+        const channels = this.channels;
+        if (!channels) {
+          return [];
+        }
+        const sortFields = ['modified'];
+        const orderFields = ['desc'];
+        const data = orderBy(
+          this.channels.filter(channel => channel[ChannelListTypes.EDITABLE] && !channel.deleted),
+          sortFields,
+          orderFields,
+        );
+        return data;
+      },
+
+      // channel items properties
+      canEdit() {
```

Comment:
No action required - just mentioning this for context since you expressed interest in some upcoming work.

Since we're fetching only editable channels, I think that strictly speaking this will always evaluate to `true` on this particular page. However we can keep it, and related conditions in this PR, because the way it's implemented will likely be handy for an upcoming task (not in this PR) where a channel card will be abstracted to `StudioChannelCard` that will have `editable` Boolean prop.